pkg/transform/kubernetes/kubernetes.go

```diff
@@ -189,14 +190,91 @@ func (t *Transformer) CreateDeployments(o *object.Service) ([]runtime.Object, er
 			})
 		}
 
+		// TODO: It is assumed that the check is done about the existence of volume in root level volume section
+		for _, mount := range c.Mounts {
+
+			volumeMount := api_v1.VolumeMount{
+				Name:      mount.VolumeName,
+				ReadOnly:  mount.ReadOnly,
+				MountPath: mount.MountPath,
+				SubPath:   mount.VolumeSubPath,
+			}
+
+			kc.VolumeMounts = append(kc.VolumeMounts, volumeMount)
+
+			// if this mount does not exist in emptydir then this is coming from root level volumes directive
+			// if tomorrow we add support for ConfigMaps or Secrets mounted as volumes the check should be done
+			// here to see if it is not coming from configMaps or Secrets
+			if !o.EmptyDirVolumeExists(mount.VolumeName) {
```

Comment:
Wouldn't be better to do `if o.VolumeExists(mount.VolumeName)` here? It might make more sense once we add secrets as volumes and It will be also a bit safer.